src/zos-pty.c

```diff
@@ -0,0 +1,108 @@
+/////////////////////////////////////////////////////////////////////////////////
+//// Licensed Materials - Property of IBM
+//// ZOSLIB
+//// (C) Copyright IBM Corp. 2022. All Rights Reserved.
+//// US Government Users Restricted Rights - Use, duplication
+//// or disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+/////////////////////////////////////////////////////////////////////////////////
+
+#define _AE_BIMODAL 1
+
+#include "zos-base.h"
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <termios.h>
+#include <pty.h>
+
+int openpty(int *master, int *slave, char *name, const struct termios *termp, const struct winsize *winp) {
+    int fd;
+    char master_dev[20], slave_dev[20];
+
+    *master = -1;
+    *slave = -1;
+
+    // Iterating through the 4-digit ptyp and ttyp devices
+    int i=0;
+    for (; i < 10000; i++) {  // Maximum 9999 for 4-digit format
+        snprintf(master_dev, sizeof(master_dev), "/dev/ptyp%04d", i);
+        snprintf(slave_dev, sizeof(slave_dev), "/dev/ttyp%04d", i);
+
+       if ((*master = open(master_dev, O_RDWR | O_NOCTTY)) != -1) {
+                if (grantpt(*master) != 0)
```

Comment:
Are you able to make the spacing match the indentation of the other files? (tabs are 2 spaces)

Reply:
Indentation corrected